odoo/tools/mail.py

```diff
@@ -259,7 +259,17 @@ def html_sanitize(src, silent=True, sanitize_tags=True, sanitize_attributes=Fals
 URL_REGEX = r'(\bhref=[\'"](?!mailto:|tel:|sms:)([^\'"]+)[\'"])'
 TEXT_URL_REGEX = r'https?://[\w@:%.+&~#=/-]+(?:\?\S+)?'
 # retrieve inner content of the link
-HTML_TAG_URL_REGEX = URL_REGEX + r'([^<>]*>([^<>]+)<\/)?'
+TEXT_BEFORE_CLOSING_TAG_REGEX = r'([^<>]*>([^<>]+)<\/)?'
+NO_TAG = r'[^<>]*'
+QUOTE = r'[\'\"]'
+NO_QUOTE = r'[^\'\"]'
+
+HTML_TAG_URL_REGEX = URL_REGEX + TEXT_BEFORE_CLOSING_TAG_REGEX
+HTML_TAG_IMAGE_ALT_REGEX = (
+        URL_REGEX
```

Comment:
the regex need to be "multiline" (re.ALL I think), otherwise this won't work
```html
<a href="https://www.odoo.com">
    <img src="https://www.odoo.com/logo.png" alt="image"/>
</a>
```

Also if there's space between the end of `<a>` and the start of `<img>` the label is not catched
```html
<a href="https://www.odoo.com"> <img src="https://www.odoo.com/logo.png" alt="image"/> </a>
```

(+ add test for the first scenario)

Reply:
Test added! All content except tags will work before the `<img` though, even on different lines without the flag as we don't use `^` or `$`.